Group related changes in RELEASE_NOTES

foundrytom opened the pull request:
Trying to make the bigger picture a bit easier to grok...

Question - should we get rid of the emoji hammer for build until we have other ones we use for other areas? So its not all on its own? (or add ones for the other things now)

elliotcmorris reviewed (approved):
:+1: 

As for the build emoji, my feeling is that it's not overly helpful without a glossary indicating what it means specifically. There's lots of interpretations.

feltech commented on RELEASE_NOTES.md:
```diff
@@ -13,21 +13,31 @@ v1.0.0-alpha.xx
+  - Post-initialization methods raise a `NotImplementedException`.
+
+  Note that in order to suport multi-language plugins, pure virtual/ABC
+  is not used to denote requierd methods in all cases. See the class API
```
[fix] Spelling
```suggestion
  is not used to denote required methods in all cases. See the class API
```

feltech commented on RELEASE_NOTES.md:
```diff
@@ -13,21 +13,31 @@ v1.0.0-alpha.xx
+
+  Note that in order to suport multi-language plugins, pure virtual/ABC
+  is not used to denote requierd methods in all cases. See the class API
+  documetnation for more details.
```
[fix] Spelling
```suggestion
  documentation for more details.
```

feltech commented on RELEASE_NOTES.md:
```diff
@@ -13,21 +13,31 @@ v1.0.0-alpha.xx
-  the `Paged` suffix.
+  the `Paged` suffix, and renamed the `PagedRelationshipSuccessCallback`
+  to `RelationshipQuerySuccessCallback` to better reflect the default of
+  pagnation, and reduce ambiguity when relationship creation methods are
```
[fix] Spelling
```suggestion
  pagination, and reduce ambiguity when relationship creation methods are
```

feltech commented on RELEASE_NOTES.md:
```diff
@@ -13,21 +13,31 @@ v1.0.0-alpha.xx
+  - Pre-initialization methods are a no-op or return empty data.
+  - Post-initialization methods raise a `NotImplementedException`.
+
+  Note that in order to suport multi-language plugins, pure virtual/ABC
```
[fix] Spelling
```suggestion
  Note that in order to support multi-language plugins, pure virtual/ABC
```

feltech commented on RELEASE_NOTES.md:
```diff
@@ -13,21 +13,31 @@ v1.0.0-alpha.xx
+  [#DR022](doc/decisions/DR022-Default-method-implementations.md),
+  [#1113](https://github.com/OpenAssetIO/OpenAssetIO/issues/1113)
 
 - Removed non-paged versions of the `Manager`/`ManagerInterface`
```
[consider][minor] There seems to be a slight overuse of ", and" in this note

foundrytom replied:
Oh man there is - sorry! How's [this](https://github.com/OpenAssetIO/OpenAssetIO/pull/1157/commits/22aa0026b64e37d6fbfb786900895a6a5bfc40f1)?

feltech commented on RELEASE_NOTES.md:
```diff
@@ -72,11 +71,16 @@ v1.0.0-alpha.xx
   exceptional conveniences.
```
[consider][minor] Could fix this one whilst we're at it "single exception type for exceptional type"

foundrytom replied:
Hows [this](https://github.com/OpenAssetIO/OpenAssetIO/pull/1157/commits/7f5d0ea833782d5883600d04bc3e9f06d80d89da)?

feltech replied:
Perfect.

feltech reviewed (approved):
Just some spelling mistakes, though I think they weren't actually introduced by this PR, just moved.

foundrytom commented:
> I don't mind the hammer emoji. I think its useful to call out somehow that something is build-only breaking change.

I've removed this for now, and added https://github.com/OpenAssetIO/OpenAssetIO/issues/1160 - so we can make the most of it for the b1 release notes.
